src/app/(app)/course/[courseId]/(hooks)/useCourse.tsx

```diff
@@ -0,0 +1,50 @@
+"use client"
+
+import { useParams } from "next/navigation"
+import { PropsWithChildren, createContext, useContext, useEffect, useState } from "react"
+
+type CourseContextType = {
+    courseId: number
+    courseName: string
+}
+
+const CourseContext = createContext<CourseContextType>({
+  courseId: 0,
+  courseName: '',
+})
+
+const useCourseProvider = (): CourseContextType => {
+  const { courseId } = useParams<{ courseId: string }>()
+  const [courseName, setCourseName] = useState<string>('')
+
+  useEffect(() => {
+    const fetchCourse = async () => {
+      try {
+        const courseResponse = await fetch(`${process.env.NEXT_PUBLIC_BACKEND_URL}/api/v1/courses/?id=${courseId}`,)
```

Comment:
wait I feel like you're calling this api endpoint wrong.
I thought django did `.../courses/1` by default, we shouldn't be using query params for this